From a64d8e7e931092c2b92e66b7ac2b690a7c442799 Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Wed, 6 Aug 2025 10:56:13 +0800
Subject: [PATCH 027/105] mtk: mt76: mt7996: Fix get inband discovery template
 for mld bss

Add link id parameter for getting correct inband discovery template.

Signed-off-by: Allen Ye <allen.ye@mediatek.com>
---
 mt7996/mcu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index f65a1b025..7f97fd408 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -2993,11 +2993,11 @@ int mt7996_mcu_beacon_inband_discov(struct mt7996_dev *dev,
 	if (changed & BSS_CHANGED_FILS_DISCOVERY &&
 	    link_conf->fils_discovery.max_interval) {
 		interval = link_conf->fils_discovery.max_interval;
-		skb = ieee80211_get_fils_discovery_tmpl(hw, vif);
+		skb = ieee80211_get_fils_discovery_tmpl(hw, vif, link_conf->link_id);
 	} else if (changed & BSS_CHANGED_UNSOL_BCAST_PROBE_RESP &&
 		   link_conf->unsol_bcast_probe_resp_interval) {
 		interval = link_conf->unsol_bcast_probe_resp_interval;
-		skb = ieee80211_get_unsol_bcast_probe_resp_tmpl(hw, vif);
+		skb = ieee80211_get_unsol_bcast_probe_resp_tmpl(hw, vif, link_conf->link_id);
 	}
 
 	if (!skb) {
-- 
2.45.2

